0000-node-requirements/0000-node-requirements.md

```diff
@@ -0,0 +1,150 @@
+
+## ILP Addresses
+
+An ILP packet is addressed using an ILP Address. The ILP Addressing scheme is similar to the IP addressing scheme (both IPv4 and IPv6) in that the scheme is a hierarchical address space where routing is done based on (primarily) longest prefix match of a destination address against the address range prefixes in the node's routing table.
+
+Unlike IP addresses, ILP addresses are variable in size and encoded as IA7 (ASCII) strings of dot (`.`) separated segments. The valid characters in an address segment are alphanumeric plus Underscore (`_`), Tilde (`~`) and Hyphen (`-`) and are case sensitive which allows base64 encoded data to be encoded into address segments.
+
+More details on ILP Addresses are in [ILP RFC 15](https://interledger.org/rfcs/0015-ilp-addresses/).
+
+## Amounts
+
+Amounts in ILP are simple and are represented as 64-bit unsigned integers. As such an amount will have an associated currency and scale which specifies how the integer representation can be converted to a more user friendly decimal form when required.
+
+For example a link between two peers on the network may be settled in US dollars and the peers may decide they wish to allow precision up to 100ths of a cent. In this case the link will be configured by both peers to use the currency USD and the scale of 4. An ILP packet with an amount of 12345 is $1.23,45 or one dollar and twenty-three point four five cents.
+
+# Requirements
+
+## Links
+
+A node must have one or more links to peer nodes with whom it will exchange ILP packets. These links can use any protocol to exchange the packets as long as both peers are able to **correctly correlate which packets are responses to a specific previous request**.
+
+When two peers establish a link they must **agree on the settlement currency and scale for packets exchanged over that link**. The amount in an ILP packet is an unsigned 64-bit number, therefor when a node receives a packet it must infer the currency and scale of the amount from the link on which it was received.
+
+An example of a protocol that can be used between nodes is the [Bilateral Transfer Protocol (BTP)](https://interledger.org/rfcs/0023-bilateral-transfer-protocol/).
+
+## Link Relations
+
+Each link will have one of three relation types which reflect how the node is related to the peer on the other side of the link, these are **peer**, **parent** or **child**. The network graph is organized in a tiered hierarchy, similar to the Internet, reflecting these relationships. Large, high volume nodes are peered with one another to form the backbone of the network. Smaller nodes will have links to these "tier 1" nodes and the link will be of type child from the perspective of the tier 1 node and of type parent from the perspective of the smaller node.
```

Comment:
The current JS connector code defines a 4th relation called `local` (see [here](https://github.com/interledgerjs/ilp-connector/blob/master/src/routing/relation.ts#L1)). Should this be mentioned?

Reply:
`local` is not really a relation because it doesn't apply to relations with other peers but it is required in the code the purposes of identifying links that route to a local account.